pkg/controller/util.go

```diff
@@ -134,22 +143,43 @@ Users:
 				case "inrole":
 					t.MemberOf = append(t.MemberOf, s)
 				default:
-					c.logger.Warningf("unknown key %q", p)
+					c.logger.Warningf("unkonwn key %q", p)
 				}
 			}
-
-			delete(data, key)
+			delete(secretData, key)
 		}
 
 		if t.Name != "" {
 			result[t.Name] = t
 		}
 	}
 
-	if len(data) != 0 {
-		c.logger.Warningf("%d unprocessed entries in the infrastructure roles' secret", len(data))
-		c.logger.Info(`infrastructure role entries should be in the {key}{id} format, where {key} can be either of "user", "password", "inrole" and the {id} a monotonically increasing integer starting with 1`)
-		c.logger.Debugf("unprocessed entries: %#v", data)
+	if len(secretData) != 0 {
+		c.logger.Debugf("Extra data in the infrastructure role, checking configmap %v", rolesSecret.Name)
+		// perhaps we have some map entries with usernames, passwords, let's check if we have those users in the configmap
+		if infraRolesMap, err := c.KubeClient.ConfigMaps(rolesSecret.Namespace).Get(rolesSecret.Name, metav1.GetOptions{}); err == nil {
+			// we have a configmap with username - json description, let's read and decode it
+			for role, s := range infraRolesMap.Data {
+				if roleDescr, err := readDecodedRole(s); err != nil {
+					// check if we have a a password in a configmap
+					return nil, fmt.Errorf("could not decode role description: %v", err)
+				} else {
+					c.logger.Debugf("found role description for role %q: %+v", role, roleDescr)
+					if passwd, ok := secretData[role]; ok {
+						roleDescr.Password = string(passwd)
+						delete(secretData, role)
+					}
+					roleDescr.Name = role
```

Comment:
This means repeating the name within the entry value's YAML is redundant and ultimately ignored, right?  I think this is better, but then it should be removed from the examples. (Unmarshalling should still work and just leave the name as an empty string, no?)

Reply:
Looking at the code I'm not sure about how "ultimately" it's being ignored - so far it's ignored only in the result of this `getInfrastructureRoles` function.